tests/forms_tests/tests/test_forms.py

```diff
@@ -5069,3 +5070,20 @@ class CustomFrameworkForm(FrameworkForm):
             '<label for="id_name" class="required">Name:</label>'
             '<legend class="required">Language:</legend>',
         )
+
+
+class DeprecationTests(SimpleTestCase):
+    def test_warning(self):
+        msg = (
+            'The template "django/forms/default.html" will be removed.'
+            "You should specify an appropriate template, such as "
+            '"django/forms/div.html" on your FORM_RENDERER, or Form or '
+            "FormSet subclass."
+        )
+        with (
+            isolate_lru_cache(get_default_renderer),
```

Comment:
This **shouldn't** be needed... 

`override_settings` triggers the `setting_changed` signal, which runs `reset_template_engines`, which calls `get_default_renderer.cache_clear` — but it's not working so... 

(`test_custom_renderer_template_name` does the equivalent. If we can work out what's going on here, that can be cleaned up too.)

Reply:
PY38 doesn't seem to like this block: 

```
======================================================================
ERROR [0.003s]: test_warning (forms_tests.tests.test_forms.DeprecationTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/pull-requests-bionic/database/postgis/label/bionic-pr/python/python3.8/tests/forms_tests/tests/test_forms.py", line 5083, in test_warning
    with (
AttributeError: __enter__
```